### DISCUSSION TITLE
FoxESS KH10.5 and Waveshare RS485

### DISCUSSION THREAD

**Discussion by lamixer** [nathanmarlor/foxess_modbus#421] [Support] [answered]
Hello! I got my Fox ESS system installed today and was prepared with a Waveshare RS485 to PoE server and had the installers wire it up. I see all three lights on the top of the Waveshare device lit up when I connect the RS485. I have followed the set up guide for the Waveshare device, mostly -- I only used the Webconfig, and it's running firmware 1.452.

When I try to set up the integration in HA I get this error:

`Error connecting to your adapter or inverter. Ensure the adapter is properly configured and is correctly wired to your inverter (see the setup link above), then try again. Details: Modbus Error: [Input/Output] Modbus Error: [Invalid Message] No response received, expected at least 8 bytes (0 received)`

Or, if I turn off multihost, I get:

`Error connecting to your adapter or inverter. Ensure the adapter is properly configured and is correctly wired to your inverter (see the setup link above), then try again. Details: Modbus Error: [Input/Output] Modbus Error: [Invalid Message] Incomplete message received, expected at least 8 bytes (1 received)`

Any suggestions come to mind? I am confused if the "Destination IP/DNS" field means anything when in TCP Server mode. I have it set to the IP of my HA installation. Thanks!

**Comment by FozzieUK** [comment]
The DestinationIP/DNS field isn't important, I set mine to point to an unused IP address in my normal IP range.

The Device IP is the important one, that's the address of the Waveshare device and the Device Port should be set to 502, web port 80 and the Gateway your default gateway.
Serial data rate 9600, N, 8, 1 and the Protocol should be set to Modbus TCP to RTU and enable multi host should be on.

What worries me is that you say all three lights came on, what I would expect is the Power & Link lights, but not the ACT light - that should flash when it is polled - if that light is on permanently it usually means that you have the A & B wires crossed.

**Reply by lamixer** [reply]
Thanks FozzieUK ... I did have the A & B reversed, figuring that light on = good. With it connected the other way, when I click to add the integration in HA I see the ACT light flash. I thought I was then going to be delighted, but still I get a similar message (0 bytes this time):

`Error connecting to your adapter or inverter. Ensure the adapter is properly configured and is correctly wired to your inverter (see the setup link above), then try again. Details: Modbus Error: [Input/Output] Modbus Error: [Invalid Message] No response received, expected at least 8 bytes (0 received)`

Hmm. I think it's set correctly. Possibly the electricians made a mistake, I'll have to check the multipin connector on the inverter when I have a chance.

**Reply by FozzieUK** [reply]
That’s a better start, if it flashes you are polling it, but sounds like you are not getting a (correct) response.

It’s worth installing modpoll.exe and testing from there, it does provide a bit more feedback ‘no reply’, Illegal Data Address etc.. 

But yes definitely check the multi pin, a common mistake is to connect it to the RS485 ‘meter’ pins, they should be connected to the 485A&B connections

**Reply by lamixer** [reply, marked as answer]
Ok... I checked the wiring, it was good. I played with everything again and got it to work -- it's amazing by the way, thank you! I think the problem was I had the "Instruction Time out" set too low, it was at 32, I put it at 512 which is working now. Also I had dthe RS485 Conflict Time Gap set to something low, maybe it was 20. I put it at 100 which is working now.

**Reply by FozzieUK** [reply]
Great, glad you got it working! - and thanks for the feedback 👍

**Reply by lamixer** [reply]
AND/OR ... I also turned on a setting in my KH-10.5 inverter called 'remote control' which was off. I had tried that before and it didn't make a difference, but it's on now. Do you know if it needs to be on? Does it affect RS485 (maybe read only versus being able to control things)?

**Reply by FozzieUK** [reply]
I’m not familiar enough with the KH to be sure, it’s listed as PC remote control and the manual implies the ability to read inverter values using this, so it does sound very much like it should be on to enable the comms.
